xayn-ai/src/ranker/mod.rs

```diff
@@ -43,4 +64,143 @@ impl Ranker {
     pub(crate) fn compute_smbert(&self, sequence: &str) -> Result<Embedding, Error> {
         self.smbert.run(sequence).map_err(Into::into)
     }
+
+    /// Ranks the given documents based on the learned user interests.
+    ///
+    /// # Errors
+    ///
+    /// Fails if no user interests are known.
+    pub(crate) fn rank(&self, documents: &mut [Document]) -> Result<(), Error> {
+        rank(
+            documents,
+            &self.user_interests,
+            self.coi_config.neighbors.get(),
```

Comment:
just as a heads up, once #375 is merged this will change to
```suggestion
            self.coi_config.neighbors(),
```